subsys/dfu/boot/mcuboot.c

```diff
@@ -34,6 +34,16 @@
 #define BOOT_HEADER_MAGIC_V1 0x96f3b83d
 #define BOOT_HEADER_SIZE_V1 32
 
+/*
+ * Fall back to slot0_partition if zephyr,code_partition chosen node is not defined.
+ * This allows building for platforms such as native_posix where flash offset cannot be set.
+ */
+#if DT_HAS_CHOSEN(zephyr_code_partition)
+#define ACTIVE_SLOT_FLASH_AREA_ID DT_FIXED_PARTITION_ID(DT_CHOSEN(zephyr_code_partition))
+#else
+#define ACTIVE_SLOT_FLASH_AREA_ID DT_FIXED_PARTITION_ID(DT_NODELABEL(slot0_partition))
+#endif
```

Comment:
Why? native_sim cannot use mcuboot, so what's the point in this?

Reply:
It can't, but the tests under `zephyr/tests/subsys/dfu` build for the `native_posix` target. Those tests set `CONFIG_IMG_MANAGER`, which compiles in `zephyr/subsys/dfu/boot/mcuboot.c`, and test (to some extent) the APIs implemented in that source file.

I agree that it doesn't really make sense to make the exception here, as as the `native_posix` target can't use MCUboot, but without the exception those tests cannot compile.

Reply:
Seems that there are fake partitions set up, interesting. This here needs to go and chosen node needs adding to native_sim's dts file - if that causes any issues with other tests then it can be applied as an overlay to the affected samples.

Reply:
@nordicjm Please take another look, I've made the suggested change and all tests pass.